src/python/pants/backend/python/dependency_inference/rules.py

```diff
@@ -514,7 +529,21 @@ async def infer_python_conftest_dependencies(
         Get(Owners, OwnersRequest((f,), OwnersNotFoundBehavior.error))
         for f in conftest_files.snapshot.files
     )
-    return InferredDependencies(itertools.chain.from_iterable(owners))
+
+    original_tgt, owner_tgts = await MultiGet(
+        Get(WrappedTarget, Address, request.sources_field.address),
+        Get(Targets, Addresses(itertools.chain.from_iterable(owners))),
+    )
+    resolve = original_tgt.target[PythonResolveField].normalized_value(python_setup)
+    python_owners = [
+        tgt.address
+        for tgt in owner_tgts
+        if (
+            tgt.has_field(PythonSourceField)
+            and tgt[PythonResolveField].normalized_value(python_setup) == resolve
```

Comment:
Should this be using `tgt.get`?

Reply:
Maybe - we use the hard check in `module_mapper.py` so nothing is new here. We'll error if plugin authors aren't considering resolves.

I think my thinking back then was that plugin authors should be reasoning about resolves? Otherwise the target will always use the default resolve, with no escape hatch.